ui/src/testing/stubs/jupyter-api-stub.ts

```diff
@@ -0,0 +1,46 @@
+import {JupyterApi, JupyterContents, Model} from 'notebooks-generated/fetch';
+
+export class JupyterApiStub extends JupyterApi {
+
+  constructor() {
+    super(undefined, undefined, (..._: any[]) => {
+      throw Error('cannot fetch in tests');
+    });
+  }
+
+  public postContents(googleProject: string, clusterName: string, workspaceDir: string,
```

Comment:
nit: I'd probably name this `getPostContents()` so it doesn't look like this function is doing a POST. Likewise for `getPutContents()`

Reply:
This overrides an API method which does a POST